ansible/roles/hotpot-installer/tasks/main.yml

```diff
@@ -13,71 +13,24 @@
 # limitations under the License.
 
 ---
-- name: install Red Hat system packages
-  package:
-    name: "{{ item }}"
-    state: present
-  with_items:
-    - librados-devel
-    - librbd-devel
-  when: ansible_os_family == "RedHat" and deploy_project != "gelato"
-
-- name: install Ubuntu system packages
-  package:
-    name: "{{ item }}"
-    state: present
-  with_items:
-    - librados-dev
-    - librbd-dev
-  when: ansible_os_family == "Debian" and deploy_project != "gelato"
-
-- name: create opensds work directory if it doesn't exist
-  file:
-    path: "{{ item }}"
-    state: directory
-    mode: 0755
+- name: generate certificates
+  shell: "{{ item }}"
   with_items:
-    - "{{ hotpot_work_dir }}"
-    - "{{ opensds_config_dir }}"
-    - "{{ opensds_driver_config_dir }}"
-    - "{{ opensds_log_dir }}"
+    - bash ./script/create_cert.sh "{{ opensds_certs_dir }}"
+  become: yes
   when: deploy_project != "gelato"
 
-- name: include scenarios/repository.yml when installing from repository
-  include: scenarios/repository.yml
-  when: install_from == "repository" and deploy_project != "gelato"
-
-- name: include scenarios/release.yml when installing from release
-  include: scenarios/release.yml
-  when: install_from == "release" and deploy_project != "gelato"
-
-- name: include scenarios/container.yml when installing from container
-  include: scenarios/container.yml
-  when: install_from == "container" and deploy_project != "gelato"
-
 - name: hotpot keystone configuration
   shell: "{{ item }}"
   with_items:
     - bash ./script/keystone.sh config hotpot
   when: opensds_auth_strategy == "keystone" and deploy_project != "gelato"
   become: yes
 
-- name: copy config templates into opensds global config folder
-  copy:
-    src: ../../../../conf/
-    dest: "{{ opensds_config_dir }}"
-  when: deploy_project != "gelato"
-
-- name: generate certificates
-  shell: "{{ item }}"
-  with_items:
-    - bash ./script/create_cert.sh "{{ opensds_certs_dir }}"
-  become: yes	
-  when: deploy_project != "gelato"
-
 - name: configure opensds global info
   shell: |
     cat >> opensds.conf <<OPENSDS_GLOABL_CONFIG_DOC
+
     [osdslet]
```

Comment:
Tthe [osdslet]  shoule be  remove as it had been added to the "roles/osdsdock/task/main.yml" file

Reply:
Hi @lijuncloud , to run osdslet daemon service, [osdslet] configuration section is required, without that section we can't run osdslet daemon. We are configuring it in controller node only.

To run osdsdock daemon service we  need [osdslet], [osdsdock], [etcd], and backend section in opensds.conf. This part will append in all dock nodes' opensds.conf

Reply:
Hi @Click2Cloud-Gamma, I am worried this could have impact with single node deployment, because both osdslet and osdsdock service would update `opensds.conf`.

Reply:
Hi @leonwanghui  I have tried this on single node installer, it is working fine.